client/src/index.ts

```diff
@@ -403,8 +396,9 @@ export class RestRepository<T> extends Repository<T> {
   async request<T>(method: string, path: string, body?: unknown): Promise<T | undefined> {
     const { databaseURL, apiKey } = this.client.options;
     const branch = await this.client.getBranch();
+    const fetchImpl = this.fetch;
```

Comment:
This is actually the fix 🤯

Reply:
Wow! Great find!

Yeah, no need to bind IMO.

Reply:
@SferaDev Could you please explain why this works vs just using this.fetch in the call?

Reply:
@ncheikh It's not a simple explanation 😢 

If you're curious I strongly suggest you to read [YDKJS 1st edition](https://github.com/getify/You-Dont-Know-JS/blob/1st-ed/this%20&%20object%20prototypes/README.md#you-dont-know-js-this--object-prototypes). 

What is weird is that it does work on node.js but not on V8 (CF Workers JS runtime), I would have to dig deeper to know the exact cause.

Reply:
Thank you @SferaDev, I will take a look!  I was a bit baffled why it wasn't working and tried several things but it wasn't obvious to me why it wasn't working for CF as it was working in other environments/engines. I wanted to open an MR for the fix but couldn't come up with a solution and was hopeful y'all would be able to sort it out and add clarity.

I very much appreciate the quick turn around and taking the time to answer my question!